### PR TITLE
WIKI-1103: No Default Wiki Template

### PR DESCRIPTION
**Pull request by nguyenthienhuong**

### CODE REVIEW

File: wiki-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java [hunk 1]
```diff
@@ -1520,4 +1506,28 @@ private void initEmotionIcons() {
...
+    for (WikiTemplatePagePlugin templatePlugin : templatePagePlugins_) {
+      if (templatePlugin != null && templatePlugin.getTemplates() != null) {
+        String wikiType = "";
+        String wikiOwner = "";
+        for (Template template : templatePlugin.getTemplates()) {
+          try {
+            wikiType = template.getWikiType();
+            wikiOwner = template.getWikiOwner();
+            Wiki wiki = new Wiki(wikiType, wikiOwner);
+            Wiki createdWiki = dataStorage.createWiki(wiki);
```
Comment (thomasdelhomenie) [inline wiki-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java:1520]: Creating wikis in the method which creates the template does not seem very good in term of design. Templates should be created when the wikis are created, not the opposite.

File: wiki-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java [hunk 2]
```diff
@@ -154,6 +154,7 @@ public WikiServiceImpl(ConfigurationManager configManager,
 
   @Override
   public void start() {
+    initTemplates();
```
Comment (thomasdelhomenie) [inline wiki-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java:157]: If this method is only called at startup, how the templates will be created for the wikis created then (for example when a new group or a new user access to its wiki) ?

Comment (nguyenthienhuong) [inline wiki-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java:157]: Yes, you are right, I met the same problem when creating new space. 
But when new space created, the function createWiki() will never be called. That's why this problem occurs.
Do you have a solution for this issue?

Comment (thomasdelhomenie) [inline wiki-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java:157]: That does not seem logic. Wiki creations should be done with the method createWiki(). Is it due to this commit : https://github.com/exodev/wiki/commit/8b0552651ab4eb029f54837f86b161a8a9a45f98 ? It creates a wiki in a read method (getWikiObject), and directly by calling the data storage layer. Creation of the wiki should be done at service level.

Comment (nguyenthienhuong) [inline wiki-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java:157]: Thanks Thomas, the root cause comes from this commit https://github.com/exodev/wiki/commit/8b0552651ab4eb029f54837f86b161a8a9a45f98.

**Pull request closed without merging**